openfisca_core/simulation_builder.py

```diff
@@ -404,7 +404,11 @@ def finalize_variables_init(self, entity):
                 # Hack to replicate the values in the persons entity
                 # when we have an axis along a group entity but not persons
                 array = np.tile(values, entity.count // len(values))
-                holder.set_input(period_value, array)
+                variable = holder.variable
+                # TODO - this duplicates the check in Simulation.set_input, but
+                # fixing that requires improving Simulation's handling of entities
+                if (variable.end is None) or (period_value.start.date > variable.end):
```

Comment:
This:
`if (variable.end is None) or (period_value.start.date > variable.end):`
should of course have been
`if (variable.end is None) or (period_value.start.date < variable.end):`
…and the error breaks all input handling for variables with an end date.